python/GafferSceneTest/CollectScenesTest.py

```diff
@@ -0,0 +1,167 @@
+		script["sphere"]["sets"].setValue( "spheres" )
+
+		script["cube"] = GafferScene.Cube()
+		script["cube"]["sets"].setValue( "cubes" )
+
+		script["group"] = GafferScene.Group()
+		script["group"]["in"][0].setInput( script["sphere"]["out"] )
+		script["group"]["in"][1].setInput( script["cube"]["out"] )
+
+		script["switch"] = GafferScene.SceneSwitch()
+		script["switch"]["in"][0].setInput( script["sphere"]["out"] )
+		script["switch"]["in"][1].setInput( script["cube"]["out"] )
+		script["switch"]["in"][2].setInput( script["group"]["out"] )
+
+		# Make an empty CollectScenes
+
+		script["collect"] = GafferScene.CollectScenes()
+		script["collect"]["in"].setInput( script["switch"]["out"] )
+
+		self.assertSceneValid( script["collect"]["out"] )
+		self.assertEqual( script["collect"]["out"].childNames( "/" ), IECore.InternedStringVectorData() )
+
+		# Configure it to collect the input scenes
+
+		script["collect"]["rootNames"].setValue( IECore.StringVectorData( [ "sphere", "cube", "group" ] ) )
+
+		script["expression"] = Gaffer.Expression()
+		script["expression"].setExpression( inspect.cleandoc(
+			"""
+			scenes = parent["collect"]["rootNames"]
+			parent["switch"]["index"] = scenes.index( context.get( "collect:rootName", "sphere" ) )
+			"""
+		) )
+
+		# Check we get what we expect
+
+		self.assertEqual( script["collect"]["out"].childNames( "/" ), IECore.InternedStringVectorData( [ "sphere", "cube", "group" ] ) )
+		self.assertSceneValid( script["collect"]["out"] )
+
+		script["subTree"] = GafferScene.SubTree()
+		script["subTree"]["in"].setInput( script["collect"]["out"] )
+
+		script["subTree"]["root"].setValue( "/sphere" )
+		self.assertScenesEqual( script["subTree"]["out"], script["sphere"]["out"] )
+
+		script["subTree"]["root"].setValue( "/cube" )
+		self.assertScenesEqual( script["subTree"]["out"], script["cube"]["out"] )
+
+		script["subTree"]["root"].setValue( "/group" )
+		self.assertScenesEqual( script["subTree"]["out"], script["group"]["out"] )
+
+		# Check the sets too
+
+		self.assertEqual( script["collect"]["out"]["setNames"].getValue(), IECore.InternedStringVectorData( [ "spheres", "cubes" ] ) )
+
+		self.assertEqual(
+			script["collect"]["out"].set( "spheres" ).value,
+			GafferScene.PathMatcher(
+				[ "/sphere/sphere", "/group/group/sphere" ]
+			)
+		)
+
+		self.assertEqual(
+			script["collect"]["out"].set( "cubes" ).value,
+			GafferScene.PathMatcher(
+				[ "/cube/cube", "/group/group/cube" ]
+			)
+		)
+
+	def testGlobals( self ) :
+
+		options = GafferScene.CustomOptions()
+		options["options"].addMember( "user:test", "${collect:rootName}" )
+		self.assertTrue( "option:user:test" in options["out"]["globals"].getValue() )
+
+		collect = GafferScene.CollectScenes()
+		collect["in"].setInput( options["out"] )
+		collect["rootNames"].setValue( IECore.StringVectorData( [ "a", "b" ] ) )
+
+		self.assertEqual( collect["out"]["globals"].getValue()["option:user:test"], IECore.StringData( "a" ) )
```

Comment:
Why should we have _a_ rather than _b_ in the _option:user:test_ ? Perhaps a comment would be helpful? Ah  the globals come from the first iteration of the collect?